### ISSUE TITLE
Feature Request - Obtaining Configuration Remotely

### ISSUE THREAD

**Issue by benhysell** [sounj142/HLSoft.BlazorWebAssembly.Authentication.OpenIdConnect#4]
tldr Possibly modify .AddBlazoredOpenIdConnect to allow not setting options and obtain them via the dependency injection system.

Issue 
Currently to configure OpenIdConnect for the Blazor Client one needs to know the settings at compile time, i.e.
```
 builder.Services.AddOptions()
            .AddAuthorizationCore()
            .AddBlazoredOpenIdConnect(options =>
            {
                options.Authority = "http://localhost:5000/";

                options.ClientId = "Client.Code";
                options.ResponseType = "code";

                options.Scope.Add("openid");
                options.Scope.Add("profile");
                options.Scope.Add("api");
            })
```

Setting the options.Authority to http://localhost:5000 is fine for development, however for deployment I need to 'do something else' to set options.Authority to a 'real url' for my application, i.e. not localhost.

In past SPA applications I'll obtain my open id configuration from the server the SPA was served from by making a fetch call back to my server to obtain my open id configuration information.  i.e. in Blazor Program.cs I can do:
```
 SpaSystemConfiguration spaSystemConfiguration = new SpaSystemConfiguration();
            using (HttpClient client = new HttpClient())
            {
                //this needs a full url...can't get a handle on the window
                HttpResponseMessage response = await client.GetAsync("https://localhost:5000/api/utilities/SpaConfiguration");
                response.EnsureSuccessStatusCode();
                string responseBody = await response.Content.ReadAsStringAsync();
                (_, spaSystemConfiguration) = responseBody.TryParseJson<SpaSystemConfiguration>();

                Console.WriteLine(spaSystemConfiguration);

            }
``` 

I can then feed in my spaSystemConfiguration that was obtained from my server into .AddBlazoredOpenIdConnenct.

This solution doesn't fix the issue, httpClient still needs a full url, so I'm back to hard coding http://localhost:5000 into my Blazor client.

Possible Solution
I'd love to be able to do something like:
```
builder.Services.AddSingleton(async p =>
            {
                
                var httpClient = p.GetRequiredService<HttpClient>();
                HttpResponseMessage response = await httpClient.GetAsync("api/utilities/SpaConfiguration");
                response.EnsureSuccessStatusCode();
                string responseBody = await response.Content.ReadAsStringAsync();
                (_, spaSystemConfiguration) = responseBody.TryParseJson<SpaSystemConfiguration>();

                var options = new OpenIdConnectOptions();
                //fill options in with results from spaSystemConfiguration

                return options;
            });
```

And have .AddBlazoredOpenIdConnect pick up those options from the dependency injection system.

**Comment by sounj142** [comment]
Hi benhysell,
I just released a new version, 0.7.0. This version has some new implementation that supports your request:
```csharp
.AddBlazoredOpenIdConnect(async (provider) =>
{
	var httpClient = provider.GetRequiredService<HttpClient>();
	var result = await httpClient.GetJsonAsync<OpenIdConnectOptions>("http://localhost:5001/Configs");
	return result;
});
```
Have a look example project Client.IdentityServer.Code.RemotelyConfig to have more detail.

**Comment by benhysell** [comment]
Thanks for the updated version!  I was able to use the newly released Blazor WebAssembly 3.2.0 Preview 3 and added a `appsettings.json` instead of using the `httpClient`.  i.e.

```
builder.Services.AddAuthorizationCore(options => { })
               .AddBlazoredOpenIdConnect(async (provider) =>
               {
                   var options = new OpenIdConnectOptions();                  
                   var configuration = provider.GetRequiredService<IConfiguration>();
                   options.Authority = configuration["SpaOptions:Authority"].ToString(); 
                   options.ClientId = configuration["SpaOptions:SpaClientId"].ToString();
                   options.ResponseType = "token id_token";
                   options.Scopes.Add("openid");
                   options.Scopes.Add("profile");
                   options.Scopes.Add("email");
                   options.Scopes.Add(configuration["SpaOptions:Scope"].ToString());
                   return options;
               })
```
I didn't have any luck using the `httpClient`...I was able to make the initial request to the IdServer, however the return response was "lost" and I was never able to fully authenticate.
